redash/models.py

```diff
@@ -548,11 +557,12 @@ def recent(cls, user_id=None, limit=20):
             where(~(Event.object_id >> None)).\
             where(Event.object_type == 'query'). \
             where(cls.is_archived == False).\
+            where(cls.access_groups.contains_any(*current_user.groups)).\
             group_by(Event.object_id, Query.id).\
             order_by(peewee.SQL("count(0) desc"))
 
-        if user_id:
-            query = query.where(Event.user == user_id)
+        if limit_to_current_user:
+            query = query.where(Event.user == current_user.id)
```

Comment:
`current_user` is a global that's only available in the context of a web request. Coupling the model layer with the web one will limit the usefulness of it.

If a user is needed, just pass it to this method (instead of passing `user_id` we can pass the user object).